### PR TITLE
[[ Bug ]] Ensure widgets are loaded in dependency order

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

File: Toolset/libraries/revideextensionlibrary.livecodescript [hunk 1]
```diff
@@ -668,6 +668,68 @@ private function __fetchModuleData pExtensionFile
    return tDataA
 end __fetchModuleData
 
+private function __extensionIsBuiltin pID
+   set the itemdelimiter to "."
+   return the number of items in pID is 3
+end __extensionIsBuiltin
+
```
Comment (runrevben) [inline Toolset/libraries/revideextensionlibrary.livecodescript:675]: Is this viable when custom domain registration such as com.devore.widget are allowed?

Comment (livecodeali) [inline Toolset/libraries/revideextensionlibrary.livecodescript:675]: No, it isn't - I wasn't sure exactly what the conditions were / will be which was why I made it a separate function so we can easily change it.

Comment (runrevben) [inline Toolset/libraries/revideextensionlibrary.livecodescript:675]: Sounds sensible. Is this something we'll need to add to the IDE when it loads widgets so it tracks where the widget was loaded from? I'm happy to review it for now but we should add this to the todo so we don't forget. Thoughts?

**Comment by runrevben** [comment]
Reviewed in the basis that Ali has added not for us to address identification of widgets when manifest updates are in place.

**Pull request merged**